microprofile-metrics/src/test/java/org/wildfly/quickstart/microprofile/metrics/MicroProfileMetricsIT.java

```diff
@@ -0,0 +1,244 @@
+     */
+    @Test
+    public void testParallelMetrics() throws Exception {
+        int n = 3;
+        WebTarget webTarget = client.target(deploymentURL.toString())
+            .path("/parallel");
+
+        List<Future<Response>> responses = new ArrayList<>(n);
+
+        for (int i = 0; i < n; i++) {
+            responses.add(webTarget.request().async().get());
+        }
+
+        // 3 parallel requests are in flight, give metrics a while to be collected
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        // check the application metrics
+        Response response = client.
+            target(managementURL.toString())
+            .path("/metrics/application")
+            .request()
+            .header("Accept", MediaType.APPLICATION_JSON)
+            .get();
+
+        Assert.assertEquals(200, response.getStatus());
+        JsonReader jsonReader = Json.createReader(new StringReader(response.readEntity(String.class)));
+        JsonObject json = jsonReader.readObject();
+
+        Assert.assertEquals("Invalid number of current parallel accesses",
+            3, json.getJsonObject("org.wildfly.quickstart.microprofile.metrics.PrimeNumberChecker.parallelAccess").getInt("current"));
+
+        // finish the started requests
+        restAppGetInvoke(client, "/parallel-finish");
+
+        for (int i = 0; i < n; i++) {
+            responses.get(i).get().close();
+        }
+    }
+
+    /**
+     * Tests that /injected-metric metrics are correctly collected.
+     */
+    @Test
+    public void testInjectedMetrics() {
+        for (int i = 0; i < 3; i++) {
+            restAppGetInvoke(client, "/injected-metric");
+        }
+
+        Response response = client.
+            target(managementURL.toString())
+            .path("/metrics/application")
+            .request()
+            .header("Accept", MediaType.APPLICATION_JSON)
+            .get();
+
+        Assert.assertEquals(200, response.getStatus());
+        JsonReader jsonReader = Json.createReader(new StringReader(response.readEntity(String.class)));
+        JsonObject json = jsonReader.readObject();
+
+        Assert.assertEquals(3, json.getInt("injectedCounter"));
+    }
+
+    /**
+     * Tests that /registry metrics are correctly collected.
+     */
+    @Test
+    public void testRegistryMetrics() {
+        restAppGetInvoke(client, "/registry");
+
+        Response response = client.
+            target(managementURL.toString())
+            .path("/metrics/application")
+            .request()
+            .header("Accept", MediaType.APPLICATION_JSON)
+            .get();
+
+        Assert.assertEquals(200, response.getStatus());
+        JsonReader jsonReader = Json.createReader(new StringReader(response.readEntity(String.class)));
+        JsonObject json = jsonReader.readObject();
+
+        Assert.assertEquals(42, json.getInt("programmaticCounter"));
```

Comment:
Could the number 42 be present as a constant, e.g in the `PrimeNumberChecker` class, it would simplify the understanding of the code IMHO